sql/src/main/java/io/crate/analyze/symbol/Aggregations.java

```diff
@@ -35,6 +39,47 @@ public static boolean containsAggregation(Symbol s) {
         return AGGREGATION_SEARCHER.process(s, null);
     }
 
+    /**
+     * @return true if the symbol is an aggregation or function which contains an aggregation or a scalar.
+     */
+    public static boolean containsAggregationOrscalar(Symbol s) {
+        return AGGREGATION_OR_SCALAR_SEARCHER.process(s, null);
+    }
+
+    /**
+     * @return true if the symbol is found in the group by, if the symbol is a scalar function all column arguments must be found in the group by.
+     */
+    public static boolean matchGroupBySymbol(Symbol s, List<Symbol> groupBy) {
+        return GROUP_BY_MATCHER.process(s, groupBy);
+    }
+
+    private static class AggregationOrScalarSearcher extends SymbolVisitor<Void, Boolean> {
+
+        @Override
+        protected Boolean visitSymbol(Symbol symbol, Void context) {
+            return false;
+        }
+
+        @Override
+        public Boolean visitFunction(Function symbol, Void context) {
+            if (symbol.info().type() == FunctionInfo.Type.AGGREGATE || symbol.info().type() == FunctionInfo.Type.SCALAR) {
+                return true;
+            } else {
```

Comment:
the only other `info().type()` is `TABLE`. I'm not sure if it makes sense to traverse the arguments in that case?